File: pascaline_tests/services_test1.pas
{******************************************************************************
*                                                                             *
*                         TESTS FOR SERVICES                                    *
*                                                                             *
*                   Copyright (C) 2001 S. A. Moore                            *
*                                                                             *
******************************************************************************}

program services_test1(output);

uses services;

var p: packed array 40 of char;

begin
  
   getenv('bark', p);
   writeln('This is exttst1: ''', p:*, '''');

end.

File: pascaline_tests/services_test1.c
/******************************************************************************
*                                                                             *
*                         TESTS FOR EXTLIB                                    *
*                                                                             *
*                   Copyright (C) 2001 S. A. Moore                            *
*                                                                             *
******************************************************************************/

#include <stdio.h>

extern void pa_getenv(char* ls, char*ds, int dsl);

void main()

{

   char s[40];

   pa_getenv("bark", s, 40);
   printf("This is exttst1: \'%s\'\n", s);

}